app/(pages)/_components/SelectButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import useSelectMode from '../_hooks/useSelectMode';
+
+const SelectButton: React.FC = () => {
+  const { toggleSelectMode } = useSelectMode();
+  const [isSelected, setIsSelected] = useState(false);
```

Comment:
I think the isSelected and setIsSelected can be part of the useSelectMode context. This means you can just say something like:

`const { selectMode, toggleSelectMode } = useSelectMode();`

This way lines 5 and 6 can be replaced with one line and all future components using the context don't have to write extra code :).